File: app/src/main/java/com/bth/running/weather/model/Weather.java
package com.bth.running.weather.model;

import java.util.ArrayList;
import java.util.List;

/**
 * @author Martin Macheiner
 *         Date: 21.09.2017.
 */

public class Weather {

    List<WeatherRecord> list;

    public Weather() {
        list = new ArrayList<>();
    }

    public void compress() {

        if (list.size() > 5) {
            List<WeatherRecord> compressed = new ArrayList<>();
            for (int i = 0; i < list.size(); i += (list.size()/5)) {
                compressed.add(list.get(i));
            }
            list = new ArrayList<>(compressed);
        }
    }

    public int getForecastCount() {
        return list.size();
    }

    public WeatherRecord getForecastFor(int day) {

        if (day < 0 && day >= getForecastCount()) {
            throw new IndexOutOfBoundsException("Day must be in range");
        }

        return list.get(day);
    }

    @Override
    public String toString() {
        String str =  "Size: " + list.size() + "\n";
        for (WeatherRecord record : list) {
            str += record.toString() + "\n";
        }

        return str;
    }
}

File: app/src/main/java/at/shockbytes/corey/running/weather/model/Weather.kt
package at.shockbytes.corey.running.weather.model

import java.util.*

/**
 * @author Martin Macheiner
 * Date: 21.09.2017.
 */

class Weather {

    private var list: List<WeatherRecord>

    val forecastCount: Int
        get() = list.size

    init {
        list = ArrayList()
    }

    fun compress() {
        if (list.size > 5) {
            val compressed = ArrayList<WeatherRecord>()
            var i = 0
            while (i < list.size) {
                compressed.add(list[i])
                i += list.size / 5
            }
            list = ArrayList(compressed)
        }
    }

    fun getForecastFor(day: Int): WeatherRecord {

        if (day in forecastCount..-1) {
            throw IndexOutOfBoundsException("Day must be in range")
        }
        return list[day]
    }

    override fun toString(): String {
        var str = "Size: " + list.size + "\n"
        for (record in list) {
            str += record.toString() + "\n"
        }

        return str
    }
}
